src/Frontend/Modules/MediaLibrary/Widgets/Base/FrontendMediaWidget.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Frontend\Modules\MediaLibrary\Widgets\Base;
+
+use Backend\Modules\MediaLibrary\Domain\MediaGroup\MediaGroup;
+use Frontend\Core\Engine\Base\Widget as FrontendBaseWidget;
+
+/**
+ * This has some methods to help for our FrontendMediaWidget
+ */
+class FrontendMediaWidget extends FrontendBaseWidget
+{
+    /**
+     * @var MediaGroup
+     */
+    protected $mediaGroup;
+
+    /**
+     * Execute the extra
+     */
+    public function execute()
+    {
+        // Will probably add some CSS/JS to the header
+        parent::execute();
+    }
+
+    /**
+     * Load data
+     */
+    protected function loadData()
+    {
+        // We are loading in the MediaGroup for a custom module
+        if (isset($this->data['group_id'])) {
+            /** @var MediaGroup $mediaGroup */
+            $this->mediaGroup = $this->get('media_library.repository.group')->getOneById(
+                $this->data['group_id']
+            );
+        } else {
+            throw new \Exception(
```

Comment:
I would switch it around to check if we need to throw an exception so we can drop the else